### PR TITLE
Refactor: oauth client config

### LINKED ISSUE #707: Switch (back) to static OAuth client config
In https://github.com/cal-itp/benefits/pull/698, we switched from having the `scope` specified under `AUTHLIB_OAUTH_CLIENTS` in settings to specifying it in the client after creation. I worry this could lead to some subtle bug, where Authlib is doing something special with the scope on client initialization.

An approach that would be more in line with Authlib's recommended setup would be to specify multiple clients in settings: one for each (verifier) use case, I guess? That would be something like `ial1` and `ial2_age`.

## Acceptance Criteria

- [ ] `scope` is specified in settings, rather than at runtime

### PR DESCRIPTION
**Pull request by thekaveman**
Closes #707 

I followed an approach adapted from this SO answer: https://stackoverflow.com/a/64174413

It made sense to consolidate all configuration values into the same place, and I went with the `AuthProvider` models / fixtures. There are no more env vars or settings in `settings.py`, but OAuth is still configured statically:

1. Fixtures specify `client_name`, `client_id`, `authority`, `scope`
2. On container start up, database built once, creating models from fixtures
3. On app start up, register an OAuth client for each `AuthProvider`
4. In views and elsewhere, create OAuth client instances from the registry with the user's `EligibilityVerifier.auth_provider`

Note I'm currently basing this on #710, so it would be good to get that one merged in first.

### CODE REVIEW

File: benefits/oauth/client.py [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
+from benefits.core.models import AuthProvider
+
 
 logger = logging.getLogger(__name__)
 
+oauth = OAuth()
 
-_OPENID_SCOPE = "openid"
 
+def client_kwargs(scope=None):
```
Comment (angela-tran) [inline benefits/oauth/client.py:17]: Would this function ever be used outside of this module? If not, I suggest prefixing with a `_`

File: benefits/oauth/client.py [hunk 2]
```diff
@@ -1,36 +1,52 @@
...
-def instance(scope=None):
+    `scope` should be a space-separated list of scopes to add.
     """
-    Get an OAuth client instance using the OAUTH_CLIENT_NAME setting.
+    scopes = ["openid", scope] if scope else ["openid"]
+    return {"code_challenge_method": "S256", "scope": " ".join(scopes)}
+
 
-    Optionally configure with a (space-separated) scope.
+def server_metadata_url(authority):
```
Comment (angela-tran) [inline benefits/oauth/client.py:27]: Same comment as for the function above this one: Would this function ever be used outside of this module? If not, I suggest prefixing with a `_`

File: benefits/oauth/apps.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from django.apps import AppConfig
+
+
+class OAuthAppConfig(AppConfig):
+    name = "benefits.oauth"
+    label = "oauth"
+    verbose_name = "Benefits OAuth"
+
+    def ready(self):
+        from .client import oauth, register_providers
```
Comment (afeld) [inline benefits/oauth/apps.py:13]: You meant to do this import in a function? If so, mind adding a comment explaining why?

**Review by angela-tran — COMMENTED** [review]
I really like that this refactor puts the client configuration on `AuthProvider` - it's much more aligned with our original reasoning for creating that model. This approach seems more robust.

File: benefits/oauth/client.py [hunk 4]
```diff
@@ -1,36 +1,52 @@
...
+    Register OAuth clients into the given registry, using configuration from AuthProvider models.
+    """
+    logger.info("Registering OAuth clients")
 
-    if scope:
-        scopes.append(scope)
+    providers = AuthProvider.objects.all()
 
-    client.client_kwargs["scope"] = " ".join(scopes)
+    for provider in providers:
```
Comment (afeld) [inline benefits/oauth/client.py:44]:
```suggestion
    # https://stackoverflow.com/a/64174413
    for provider in providers:
```

File: benefits/oauth/client.py (hunk 1 above)
Comment (thekaveman) [reply benefits/oauth/client.py:17]: You're right, it probably would not! 👍

File: benefits/oauth/client.py (hunk 2 above)
Comment (thekaveman) [reply benefits/oauth/client.py:27]: 👍

**Review by afeld — CHANGES_REQUESTED** [review]
A couple little things. Nice work!

File: benefits/oauth/client.py (hunk 4 above)
Comment (thekaveman) [reply benefits/oauth/client.py:44]: Rebasing, will add this.

**Review by angela-tran — DISMISSED** [review]
💯

**Comment by thekaveman** [comment]
@angela-tran rebased on `dev` 

Also raising `Exception` in `oauth` views if the client is not configured (related to my earlier trouble with registration, helps 
know about a missing client earlier and be more explicit about it)

**Comment by angela-tran** [comment]
> Also raising Exception in oauth views if the client is not configured (related to my earlier trouble with registration, helps
know about a missing client earlier and be more explicit about it)

👍 This will definitely be more helpful for troubleshooting

**Review by angela-tran — APPROVED** [review]

**Review by afeld — APPROVED** [review]

**Pull request merged**